modifierGui.py

```diff
@@ -0,0 +1,94 @@
+import re
+import json
+from alpacaModifier import AlpacaModifier
+
+# Can be played with.
+MIN_NUMBER_OF_ITEMS_IN_LIST = 4
+# There are quite a few questions of type: 'Sort the following list in alphabetical order.' We usually don't want to format those items.
+CHECK_IF_SORT_QUESTION = True
+# Check if there is a number in the instruction. Using numbered lists helps the LLM correctly answer queries that want a specific number of answers.
+CREATE_NUMBERED_LIST_IF_NUMBER_IN_INSTRUCTION = True
+NUM_MAP = {'one': 1, 'two': 2, 'three': 3, 'four': 4, 'five': 5, 'six': 6, 'seven': 7, 'eight': 8, 'nine': 9, 'ten': 10}
+
+SAVE_FILE = 'alpaca_new.json'
+
+class MyAlpacaModifier(AlpacaModifier):
+	def fill_values(self, instruction='', input='', old_output='', modified_output=''):
+		self.instruction = instruction
+		self.input = input
+		self.old_output = old_output
+		self.modified_output = modified_output
+
+	def save_prev_item(self):
+		if self.prev_item != None:
+			self.prev_item['instruction'] = self.instruction
+			self.prev_item['input'] = self.input
+			self.prev_item['output'] = self.modified_output
+			print(f'saving as:\n{self.prev_item}')
+			self.prev_item = None
+
+	def next_generator(self):
+		for item in self.data:
+			self.save_prev_item()
+			if CHECK_IF_SORT_QUESTION and 'sort' in item['instruction'].lower():
+				continue
+			
+			match = re.search(self.pattern, item['output'])
+			if match:
+				self.fill_values(item['instruction'], item['input'], item['output'], self.modify_output(match, item))
+				self.prev_item = item
+				yield self.instruction, self.input, self.old_output, self.modified_output
+
+		self.save_prev_item()
+		self.fill_values(instruction='End of matches reached.')
+
+		while True:
+			yield self.instruction, self.input, self.old_output, self.modified_output
+
+	def next_callback(self, instruction='', input='', old_output='', modified_output=''):
+		self.fill_values(instruction, input, old_output, modified_output)
+		return self.generator.__next__()
+		
+	def save_callback(self, instruction='', input='', old_output='', modified_output=''):
+		self.fill_values(instruction, input, old_output, modified_output)
+		self.save_prev_item()
+		
+		print(f"Saving modified data to {SAVE_FILE}...", end = '')
+		
+		with open(SAVE_FILE, 'w') as f:
+			json.dump(self.data, f, indent = 4)
+
+		print(f" Done.")
+		
+	def modify_output(self, match, item):
```

Comment:
You can automatically modify the captured entry here. Alternatively, you can just return the output unmodified and then manually edit it in the GUI.